pkg/restclient/glustershd.go

```diff
@@ -3,20 +3,19 @@ package restclient
 import (
 	"fmt"
 	"net/http"
-)
-
-// GlusterShdEnable sends request to api to  enable shd.
-func (c *Client) GlusterShdEnable(volname string) error {
 
-	url := fmt.Sprintf("/v1/volumes/%s/heal/enable", volname)
-	err := c.post(url, nil, http.StatusOK, nil)
-	return err
-}
-
-// GlusterShdDisable sends request to api to disable shd.
-func (c *Client) GlusterShdDisable(volname string) error {
+	"github.com/gluster/glusterd2/pkg/glustershd/api"
+)
 
-	url := fmt.Sprintf("/v1/volumes/%s/heal/disable", volname)
-	err := c.post(url, nil, http.StatusOK, nil)
-	return err
+// SelfHealInfo sends request to heal-info API
+func (c *Client) SelfHealInfo(params ...string) ([]api.BrickHealInfo, error) {
+	var url string
+	if len(params) == 1 {
+		url = fmt.Sprintf("/v1/volumes/%s/heal-info", params[0])
+	} else {
+		url = fmt.Sprintf("/v1/volumes/%s/%s/heal-info", params[0], params[1])
+	}
```

Comment:
You should also handle other cases, such as:

* no params passed
* more than 2 params passed

....in which cases, the caller should be returned an error.